genesyscloud/outbound_callabletimeset/resource_genesyscloud_outbound_callabletimeset.go

```diff
@@ -0,0 +1,139 @@
+package outbound_callabletimeset
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	"terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+
+	gcloud "terraform-provider-genesyscloud/genesyscloud"
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v119/platformclientv2"
+)
+
+/*
+The resource_genesyscloud_outbound_callabletimeset.go contains all of the methods that perform the core logic for a resource.
+*/
+
+// getAllOutboundCallableTimesets retrieves all of the Outbound Callable Timesets via Terraform in the genesys cloud and is used for the exporter
+func getAllOutboundCallableTimesets(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(resourceExporter.ResourceIDMetaMap)
+	proxy := getOutboundCallabletimesetProxy(clientConfig)
+
+	callabletimesets, getErr := proxy.getAllOutboundCallableTimeset(ctx)
+	if getErr != nil {
+		return nil, diag.Errorf("Failed to get page of callable timeset configs: %v", getErr)
+	}
+	for _, callabletimesets := range *callabletimesets {
+		resources[*callabletimesets.Id] = &resourceExporter.ResourceMeta{Name: *callabletimesets.Name}
+	}
+	return resources, nil
+}
+
+// createOutboundCallabletimeset is used by the outbound_callabletimeset resource to create Outbound Callable Timesets
+func createOutboundCallabletimeset(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := newOutboundCallableTimesetProxy(sdkConfig)
+
+	callableTimeset := platformclientv2.Callabletimeset{
```

Comment:
Use `getOutboundCallableTimesetFromResourceData` here like in the update function